### PR TITLE
Fill order checks

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: octobot_trading/personal_data/orders/order.py [hunk 1]
```diff
@@ -484,12 +484,23 @@ def update_quantity_with_order_fees(self, other_order):
...
+        Implement if necessary
+        """
+
+    async def _adapt_chained_order_before_creation(self, chained_order):
+        can_be_created = True
+        if chained_order.update_with_triggering_order_fees:
+            can_be_created = chained_order.update_quantity_with_order_fees(self)
+        # ensure price is not outdated
+        await chained_order.update_price_if_outdated()
+        return can_be_created
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/order.py:498]: Is there any reason to put this method in order.py ? it seems that it doesn't require self

File: octobot_trading/personal_data/orders/order_util.py [hunk 2]
```diff
@@ -551,6 +551,47 @@ def generate_order_id():
...
+
+
+def update_order_filled_values(order, ideal_price: decimal.Decimal):
+    if not order.filled_price:
+        # keep order.filled_price if already set (!= 0)
+        order.filled_price = _get_valid_filled_price(order, ideal_price)
+    if not order.filled_quantity or order.exchange_manager.trader.simulate:
+        # keep self.filled_quantity if already set (!= 0) in real trading
+        order.filled_quantity = order.origin_quantity
+    order.update_total_cost()
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/order_util.py:592]: I think this method should be placed in order.py to use self

File: octobot_trading/personal_data/orders/types/limit/limit_order.py [hunk 3]
```diff
@@ -36,6 +38,38 @@ def update_from_raw(self, raw_order):
...
+    def _update_limit_price_if_necessary(self, current_price):
+        updated_price = self.origin_price
+        if self.side is enums.TradeOrderSide.BUY:
+            highest_accepted_buy_price = (
+                current_price * (constants.ONE + constants.CHAINED_ORDERS_OUTDATED_PRICE_ALLOWANCE)
+            )
+            if self.origin_price > highest_accepted_buy_price:
+                # buy price is more than CHAINED_ORDERS_OUTDATED_PRICE_ALLOWANCE % higher than the current price
+                # => Reduce it to the highest allowed price
+                updated_price = highest_accepted_buy_price
```
Comment (Herklos) [inline octobot_trading/personal_data/orders/types/limit/limit_order.py:60]: :+1:

File: tests/personal_data/orders/types/limit/test_limit_order.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+
+        # with outdated price
+        buy_limit_order.exchange_manager.exchange_symbols_data.get_exchange_symbol_data(buy_limit_order.symbol).\
+            prices_manager.set_mark_price(decimal.Decimal("90"), enums.MarkPriceSources.EXCHANGE_MARK_PRICE.value)
+        await buy_limit_order.update_price_if_outdated()
+
+        _update_limit_price_if_necessary_mock.assert_called_once()
+        # price got adapted
+        assert buy_limit_order.origin_price < buy_order_price
+        assert buy_limit_order.origin_price == decimal.Decimal("90") * (
```
Comment (Herklos) [inline tests/personal_data/orders/types/limit/test_limit_order.py:76]: :+1:

File: octobot_trading/personal_data/orders/order.py (hunk 1 above)
Comment (GuillaumeDSM) [reply octobot_trading/personal_data/orders/order.py:498]: indeed, i'll move it to order_util 👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**